udf/iso8601_str_to_date.sql

```diff
@@ -0,0 +1,23 @@
+/*
+
+Take a ISO 8601 date or date and time string and return a DATE.  Return null if parse fails.
+
+Possible formats: 2019-11-04, 2019-11-04T21:15:00+00:00, 2019-11-04T21:15:00Z, 20191104T211500Z
+
+ */
+
+CREATE TEMP FUNCTION udf_iso8601_str_to_date(date_str STRING) RETURNS DATE AS (
```

Comment:
nit: I think this should be named `parse_iso8601_date`, to be more consistent with the name of the function it's wrapping